doc/design/doi-generation-1.md

```diff
@@ -0,0 +1,38 @@
+# Publish process
+
+This process is predicated on a few tasks that are expected to receive valid input and assuming that only certain errors could happen that are out of our control.
+
+- publish enabled only if dandiset is valid (so all validation checks have passed)
+- on publish trigger:
+   - create the versioned release in the db (this prevents any asset/blob belonging to the release from being deleted)
+   - insert all the relevant metadata (version, doi, url, manifest location/checksum) into the metadata record of the new version 
+  - send this new metadata record to mint the doi (there should be no reason why this should fail unless the external resource fails. if it fails notify admins since this is likely a new bug)
+   - write out to s3 bucket (even on any datacite failure)
+   - switch over ownership of the release to dandi (unless releases are not modifiable by the dandiset creator)
```

Comment:
@satra Can you clarify this? "Owners" in `dandi-api` means a set of users who allowed to make changes through the API to a dandiset. Ownership only applies to dandisets as a whole, not versions. There's also "ownership" of the objects in the S3 bucket, but all objects belong to the sponsored bucket so I don't think that's what is meant here.

Do you mean that owners should not be allowed to modify published versions through the API?

Reply:
correct. the only people who should be able to call PATCH/DELETE/PUT on non-draft dandisets are admins.